### PR TITLE
PA-4662 | Onboard to devops-pipeline and create Service Image

### PR DESCRIPTION
**Pull request by aniketasalkar**
**Motivation**

Onboard to devops-pipeline and create Service Image

**Proposed changes**
Add pipeline.yaml and Dockerfile.ci

**Alternatives considered**

**Testing steps**

No vulnerabilities found:
https://cloudbees.ais.acquia.io/devops-pipeline-4-jenkins/blue/organizations/jenkins/DEVOPS-blackbox-exporter-PIPELINE/detail/PR-1/10/pipeline/413
image

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,10 @@
-ARG ARCH="amd64"
-ARG OS="linux"
+ARG ARCH
+ARG OS
```
Comment (goutamjaiswal) [inline Dockerfile:2]: Donot change it. Although you are passing ARG from pipeline.yaml. It will already override it

File: Dockerfile [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 FROM quay.io/prometheus/busybox-${OS}-${ARCH}:latest
 LABEL maintainer="The Prometheus Authors <prometheus-developers@googlegroups.com>"
 
-ARG ARCH="amd64"
-ARG OS="linux"
+ARG ARCH
```
Comment (goutamjaiswal) [inline Dockerfile:6]: Donot change it. Although you are passing ARG from pipeline.yaml. It will already override it

**Review by goutamjaiswal — COMMENTED** [review]
Just minor suggestions

File: .acquia/Dockerfile.ci [hunk 3]
```diff
@@ -0,0 +1,6 @@
+ARG BASE_IMAGE=quay.io/prometheus/golang-builder:1.23-base
+
+FROM $BASE_IMAGE
+
+ARG ARCH
```
Comment (goutamjaiswal) [inline .acquia/Dockerfile.ci:5]: Also I can't this ARG ae used anywhere in Dockerfile.ci

Comment (aniketasalkar) [reply .acquia/Dockerfile.ci]: Yes, we are not using this. I have removed them.

File: Dockerfile (hunk 2 above)
Comment (aniketasalkar) [reply Dockerfile]: Kept Dockerfile as is from from the source repo.

File: .acquia/pipeline.env [hunk 4]
```diff
@@ -0,0 +1,2 @@
+ARCH=amd64
+OS=linux
```
Comment (amanjoshicodes) [inline .acquia/pipeline.env:2]: Newline at EOF

File: .acquia/pipeline.yaml [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      context: "."
+      build_args:
+        - "ARCH=${ARCH}"
+        - "OS=${OS}"
+
+# This step performs container image security scan in Jfrog Xray via Artifactory
+# Build will fail whenever a vulnerability is found unless set to true
+security_scan:
+  scanner: orca
+  ignore_failures: false
```
Comment (amanjoshicodes) [inline .acquia/pipeline.yaml:38]: Newline at EOF

File: .acquia/Dockerfile.ci [hunk 6]
```diff
@@ -0,0 +1,3 @@
+ARG BASE_IMAGE=quay.io/prometheus/golang-builder:1.23-base
```
Comment (amanjoshicodes) [inline .acquia/Dockerfile.ci:1]: Use REPO_LOCATION

File: .acquia/pipeline.yaml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+team: platauto
+group: platform
+service: blackbox-exporter
+
+# Validate the formatting of the pipeline.yaml file.
+validate_config: true
+
+environment_image:
+  file: ".acquia/Dockerfile.ci"
+  context: "."
```
Comment (amanjoshicodes) [inline .acquia/pipeline.yaml:11]: Add cache_from

File: .acquia/pipeline.yaml [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+      - |
+        make common-build PREFIX=.build/${OS}-${ARCH}
+        mv .build/${OS}-${ARCH}/devops-blackbox-exporter .build/${OS}-${ARCH}/blackbox_exporter
+
+build:
+  service_image:
+    - name: devops-blackbox-exporter
+      file: "Dockerfile"
+      context: "."
+      build_args:
```
Comment (amanjoshicodes) [inline .acquia/pipeline.yaml:30]: Add cache_from

**Review by amanjoshicodes — CHANGES_REQUESTED** [review]
Did we do any updates in the image? How were the vulnerabilities resolved?

cc: @goutamjaiswal , @AbhishekRatnawat , @SaranshIam

**Comment by aniketasalkar** [comment]
> Did we do any updates in the image? How were the vulnerabilities resolved?
> 
> cc: @goutamjaiswal , @AbhishekRatnawat , @SaranshIam

We didn't update anything in the image. I guess it resolved from source itself.

**Review by AbhishekRatnawat — CHANGES_REQUESTED** [review]

File: .acquia/pipeline.yaml [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+      cache_from: devops-blackbox-exporter
+      build_args:
+        - "ARCH=${ARCH}"
+        - "OS=${OS}"
+
+# This step performs container image security scan in Jfrog Xray via Artifactory
+# Build will fail whenever a vulnerability is found unless set to true
+security_scan:
+  scanner: orca
+  ignore_failures: false
```
[the remark this answers is not in the document]
Comment (AbhishekRatnawat) [reply .acquia/pipeline.yaml:38]: Leave it, tag job will promote the image to prod and we will always image tag as git tag in sre-blackbox.

File: .acquia/Dockerfile.ci [hunk 10]
```diff
@@ -0,0 +1,4 @@
+ARG BASE_IMAGE=golang:1.23
```
Comment (goutamjaiswal) [inline .acquia/Dockerfile.ci:1]:
```suggestion
ARG BASE_IMAGE=devops-blackbox-exporter/prometheus/golang-builder:1.23-base
```

File: .acquia/pipeline.yaml (hunk 5 above)
Comment (aniketasalkar) [reply .acquia/pipeline.yaml]: Updated.

File: .acquia/pipeline.env (hunk 4 above)
Comment (aniketasalkar) [reply .acquia/pipeline.env]: Updated.

File: .acquia/pipeline.yaml (hunk 7 above)
Comment (aniketasalkar) [reply .acquia/pipeline.yaml:12]: Updated.

**Review by goutamjaiswal — APPROVED** [review]

**Review by goutamjaiswal — APPROVED** [review]
LGTM

File: .acquia/Dockerfile.ci [hunk 11]
```diff
@@ -0,0 +1,4 @@
+ARG BASE_IMAGE=devops-blackbox-exporter/prometheus/golang-builder:1.23-base
```
Comment (SaranshIam) [inline .acquia/Dockerfile.ci:1]: You can pass BASE_IMAGE as part of build_args and use it here.

Comment (aniketasalkar) [reply .acquia/Dockerfile.ci]: Updated it.

**Review by amanjoshicodes — APPROVED** [review]
LGTM

**Pull request merged**